src/api/index.js

```diff
@@ -0,0 +1,272 @@
+import config from '../config'
+import {mergeRecursive} from "../helpers/mergeObjects";
+import redirect from "react-router/es/Redirect";
+
+
+class ApiClass {
+    constructor() {
+        this.url = `${config.apiUrl}`;
+    }
+
+    makeRequest = ({url, data, token}) => {
+        const _data = {
```

Comment:
In most your requests you are using GET method.
To avoid  code duplication it could be better to set it in default config for fetch method